src/components/CustomTextInput.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { ReactElement, useState } from 'react'
+import { TextInputProps } from 'react-native'
+import { widthPercentageToDP as wp } from 'react-native-responsive-screen'
+import styled, { useTheme } from 'styled-components/native'
+
+import { CloseIcon } from '../../assets/images'
+import theme from '../constants/theme'
+import { ContentError } from './text/Content'
+
+const LINE_HEIGHT = 30
+const MIN_HEIGHT = 60
+
+const StyledTextInput = styled.TextInput`
+  font-size: ${props => props.theme.fonts.defaultFontSize};
+  font-weight: ${props => props.theme.fonts.lightFontWeight};
+  letter-spacing: ${props => props.theme.fonts.listTitleLetterSpacing};
+  font-family: ${props => props.theme.fonts.contentFontRegular};
+  color: ${prop => prop.theme.colors.primary};
+  flex: 1;
+`
+
+const ClearContainer = styled.TouchableOpacity`
+  width: ${wp('6%')}px;
+  height: ${wp('6%')}px;
+`
+
+const TextInputContainer = styled.View<{ lines: number; borderColor: string; showErrorValidation: boolean }>`
+  border: 1px solid ${props => props.borderColor};
+  padding: 0 ${props => props.theme.spacings.sm};
+  border-radius: 2px;
+  height: ${props => (props.lines > 1 ? props.lines * LINE_HEIGHT : MIN_HEIGHT)}px;
+  flex-direction: row;
+  margin-bottom: ${props => (props.showErrorValidation ? 0 : props.theme.spacings.xs)};
+`
+
+const IconContainer = styled.View<{ multiLine: boolean }>`
+  align-self: ${props => (props.multiLine ? 'flex-start' : 'center')};
+  padding: ${props => props.theme.spacings.xs} 0;
+`
+
+const ErrorContainer = styled.View`
+  min-height: ${props => props.theme.spacings.lg};
+`
+
+interface CustomTextInputProps extends TextInputProps {
+  value: string
+  onChangeText: (value: string) => void
+  lines?: number
+  clearable?: boolean
+  rightContainer?: ReactElement
+  errorMessage?: string
+  customBorderColor?: string
+}
+
+// depending on error and focus the border color will be defined
+const getBorderColor = (hasErrorMessage: boolean, isFocused: boolean): string => {
+  if (hasErrorMessage) {
+    return theme.colors.incorrect
+  }
+  return isFocused ? theme.colors.primary : theme.colors.textSecondary
+}
+
+const CustomTextInput = ({
+  value,
+  onChangeText,
+  lines = 1,
+  clearable = false,
+  placeholder,
+  textContentType,
+  rightContainer,
+  errorMessage,
+  editable = true,
+  onSubmitEditing,
+  customBorderColor
+}: CustomTextInputProps): ReactElement => {
+  const theme = useTheme()
+  const [isFocused, setIsFocused] = useState<boolean>(false)
+  const hasErrorMessage = !!(errorMessage && errorMessage.length > 0)
+  const showErrorValidation = errorMessage !== undefined
```

Comment:
Do we really need both? I think this is quite similar and I think we don't have to handle `errorMessage.length === 0` specifically.

Reply:
yes we have because we render the container as a placeholder if there is no error message to reserve the height for an error message

Reply:
I also think this should be adjusted here.
1. Why are you testing for undefined one time with `!!errormessage` and one time with `errorMessage !== undefined`
2. Why at all introducing the hasErrorMessage variable which is only used once

Renaming of `showErrorValidation` would also make sense, so it is obvious what the variable does. 

I think it is not really a good design, that an empty string passing has a meaning on the design.

Reply:
1. we can use in both cases `errorMessage !== undefined` if i just use `errorMessage` its false for an empty string and the ErrorContainer (Placeholder) will not be displayed
2.  removed var

I understand the opinion that an empty string shouldn't change the design. So what you suppose, a separate boolean var like `hasErrorValidation` or sth similar? And then pass two separate props?